api/infoGatherer/specDetails.go

```diff
@@ -59,6 +60,11 @@ type specsCache struct {
 	specDetails map[string]*SpecDetail
 }
 
+type paramsCache struct {
+	mutex  sync.RWMutex
+	params map[string][]gauge.StepArg
```

Comment:
Since the parameters will be exclusively either `static` or `dynamic`, it may be worth maintaining them as separate cache.